src/systems/a320_systems/src/electrical/alternating_current.rs

```diff
@@ -0,0 +1,503 @@
+    electrical::{
+        combine_potential_sources, CombinedPotentialSource, Contactor, ElectricalBus,
+        ElectricalBusType, EmergencyGenerator, EngineGenerator, ExternalPowerSource, Potential,
+        PotentialSource, PotentialTarget, TransformerRectifier,
+    },
+    engine::Engine,
+    shared::DelayedTrueLogicGate,
+    simulation::{SimulationElement, SimulationElementVisitor, UpdateContext},
+};
+use uom::si::{f64::*, velocity::knot};
+
+pub(super) struct A320AlternatingCurrentElectrical {
+    main_power_sources: A320MainPowerSources,
+    ac_ess_feed_contactors: A320AcEssFeedContactors,
+    ac_bus_1: ElectricalBus,
+    ac_bus_2: ElectricalBus,
+    ac_ess_bus: ElectricalBus,
+    ac_ess_shed_bus: ElectricalBus,
+    ac_ess_shed_contactor: Contactor,
+    tr_1: TransformerRectifier,
+    tr_2: TransformerRectifier,
+    tr_ess: TransformerRectifier,
+    ac_ess_to_tr_ess_contactor: Contactor,
+    emergency_gen: EmergencyGenerator,
+    emergency_gen_contactor: Contactor,
+    static_inv_to_ac_ess_bus_contactor: Contactor,
+    ac_stat_inv_bus: ElectricalBus,
+}
+impl A320AlternatingCurrentElectrical {
+    pub fn new() -> Self {
+        A320AlternatingCurrentElectrical {
+            main_power_sources: A320MainPowerSources::new(),
+            ac_ess_feed_contactors: A320AcEssFeedContactors::new(),
+            ac_bus_1: ElectricalBus::new(ElectricalBusType::AlternatingCurrent(1)),
+            ac_bus_2: ElectricalBus::new(ElectricalBusType::AlternatingCurrent(2)),
+            ac_ess_bus: ElectricalBus::new(ElectricalBusType::AlternatingCurrentEssential),
+            ac_ess_shed_bus: ElectricalBus::new(ElectricalBusType::AlternatingCurrentEssentialShed),
+            ac_ess_shed_contactor: Contactor::new("8XH"),
+            tr_1: TransformerRectifier::new(1),
+            tr_2: TransformerRectifier::new(2),
+            tr_ess: TransformerRectifier::new(3),
+            ac_ess_to_tr_ess_contactor: Contactor::new("15XE1"),
+            emergency_gen: EmergencyGenerator::new(),
+            emergency_gen_contactor: Contactor::new("2XE"),
+            static_inv_to_ac_ess_bus_contactor: Contactor::new("15XE2"),
+            ac_stat_inv_bus: ElectricalBus::new(
+                ElectricalBusType::AlternatingCurrentStaticInverter,
+            ),
+        }
+    }
+
+    #[allow(clippy::too_many_arguments)]
+    pub fn update<T: ApuGenerator>(
+        &mut self,
+        context: &UpdateContext,
+        engine1: &Engine,
+        engine2: &Engine,
+        apu: &AuxiliaryPowerUnit<T>,
+        ext_pwr: &ExternalPowerSource,
+        hydraulic: &A320Hydraulic,
+        overhead: &A320ElectricalOverheadPanel,
+    ) {
+        self.emergency_gen.update(
+            // ON GROUND BAT ONLY SPEED <= 100 kts scenario. We'll probably need to move this logic into
+            // the ram air turbine, emergency generator and hydraulic implementation.
+            hydraulic.is_blue_pressurised()
```

Comment:
In hydraulics simulation to come this is a threshold of blues_pressure > 300psi. Hope it won't break anything there until it is correctly modeled in hydraulics